sympy/polys/rings.py

```diff
@@ -2531,15 +2531,140 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, x, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the expression for.
+        x : generator or generator index
+            The generator or generator index to compute the expression for.
+        dg : int
+            The degree of the monomial to compute the expression for.
+
+        Returns
+        =======
+
+        sympy.Expr
+            The expression of the polynomial ``f`` at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> dg = 2
+        >>> p.coeff_wrt(2, dg) # Using the generator index
+        10*x + 10
+        >>> p.coeff_wrt(z, 2) # Using the generator
+        10*x + 10
+
+        """
+        p = self
+        i = p.ring.index(x)
+        terms = [(m, c) for m, c in p.iterterms() if m[i] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:i] + (0,) + m[i + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem(self, g, x=None):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
+
+        The function `prem` returns the pseudo-remainder `r` such that `f` and `g`, the
+        pseudo-remainder `r` of `f` with respect to `g` is a polynomial such that there
+        exist polynomials `q0`, `q1`, ..., `q(n-1)` such that:
+            `f = q0 * g + r`
+
+        where the degree of `r` is less than the degree of `g`.
```

Comment:
This is unclear and ambiguous.

There are references to `f` but there is no parameter `f`.

The definition should also refer to the parameter `x`.

It is confusing to refer to the polynomials `q0`, ... which are not at all defined.

Also degree here means degree wrt `x` so that should be stated explicitly.

Reply:
``` python
    The pseudo-remainder `r` of polynomial `self` with respect to `g` is a polynomial
    such that there exist polynomials `q0`, `q1`, ..., `q(n-1)` satisfying the equation:
    
        self = q0 * g + r
        
    where the degree of polynomial `r` (with respect to the variable `x`) is strictly
    lower than the degree of polynomial `g` (with respect to the variable `x`).
```
Is it clear ?

Reply:
> Is it clear ?

No, because you haven't said anything about what the polynomials `q0` etc are. This definition does not mean anything mathematically.

The definition in the Maple docs is much clearer:
https://www.maplesoft.com/support/help/maple/view.aspx?path=prem

Here is a longer explanation although we should use something shorter than this:

Given two univariate polynomials `f` and `g` with coefficients in a field we can define Euclidean polynomial division uniquely. The Euclidean division of `f` by `g` is a pair of polynomials `q` and `r` (the quotient and remainder respectively) such that
$$f = gq + r$$
where `r` satisfies a degree bound: $\mathrm{deg}(r) < \mathrm{deg}(g)$.

In general Euclidean division of polynomials requires arbitrary exact division in the coefficient ring so for Euclidean division to be always well defined we require the coefficients to belong to a field (rather than a general ring). In particular we will in general need to be able to divide by the leading coefficient of `g`, $\mathrm{LC}(g)$ or otherwise the Euclidean division might not exist for all polynomials `f` and `g`.

Given two univariate polynomials `f` and `g` with coefficients in a domain (rather than a field) we can always define polynomial pseudo-division uniquely. The pseudo division of `f` by `g` is a pair of polynomials `q` and `r` such that
$$mf = gq + r$$
where $\mathrm{deg}(r) < \mathrm{deg}(g)$ and `m` is a power of the leading coefficient of `g`: $m = \mathrm{LC}(g)^{\mathrm{deg}(f) - \mathrm{deg}(g) + 1}$. Premultiplying `f` by `m` ensures that any divisions of the coefficients needed will be exact even though the coefficients do not belong to a field.

The `prem` method here treats multivariate polynomials in a ring like `R[x,y,z]` as univariate polynomials with coefficients that are polynomials e.g. like an element of the ring `R[x,y][z]`. Then we can say that the pseudo quotient and pseudo remainder with respect to `z` when dividing `f` by `g` are polynomials `q` and `r` such that
$$mf = gq + r$$
with $\mathrm{deg}(r, z) < \mathrm{deg}(g, z)$ and $m = \mathrm{LC}(g, z)^{\mathrm{deg}(f, z) - \mathrm{deg}(g, z) + 1}$. The `prem` method returns the pseudoremainder `r`. The `pquo` method returns the pseudoquotient `q` and `pdiv` returns the tuple `(q, r)`.